src/shell/lean.cpp

```diff
@@ -567,27 +576,51 @@ int main(int argc, char ** argv) {
             return ok ? 0 : 1;
         }
 
-        std::vector<std::pair<module_id, std::shared_ptr<module_info const>>> mods;
+        struct input_mod {
+            module_id m_id;
+            std::shared_ptr<module_info const> m_mod_info;
+            log_tree::node m_lt_node;
+        };
+        std::vector<input_mod> mods;
         for (auto & mod : module_args) {
+            auto lt_node = lt_root.mk_child(mod, "", {}, log_tree::DefaultLevel, true);
+            scope_log_tree lt(lt_node);
+            mod_mgr.set_log_tree(lt_node);
             auto mod_info = mod_mgr.get_module(mod);
-            mods.push_back({mod, mod_info});
+            mods.push_back({mod, mod_info, lt_node});
```

Comment:
Ahh, that's much better. I thought you were referring to the same-named field in a previous revision.